update_prevalence.py

```diff
@@ -1604,10 +1604,27 @@ def main() -> None:
     for country in sorted(data.countries.values(), key=namegetter):
         app_locations[country.app_key] = country.as_app_data()
         for state in country.states.values():
-            if state.app_key not in app_locations:
-                app_locations[state.app_key] = state.as_app_data()
-            for county in state.counties.values():
-                app_locations[county.app_key] = county.as_app_data()
+            try:
+                state_data: Optional[AppLocation] = None
+                county_data: Dict[str, AppLocation] = {}
+                if state.app_key not in app_locations:
+                    state_data = state.as_app_data()
+                for county in state.counties.values():
+                    county_data[county.app_key] = county.as_app_data()
+                # once everything succeeded, go ahead and attach
+                if state_data is not None:
+                    app_locations[state.app_key] = state_data
+                for key, appdata in county_data.items():
+                    app_locations[key] = appdata
+            except ValueError as e:
+                print_and_log_to_sentry(f"Couldn't calculate {state}: {e}")
+
+                # calculator expects all subdivisions to have data populated
+                if state.app_key in app_locations[country.app_key].subdivisions:
+                    print_and_log_to_sentry(
+                        f"Removing {state} from app_locations[{country.app_key}].subdivisions"
+                    )
```

Comment:
The original issue that motivated this was https://github.com/microCOVID/microCOVID/issues/1570, which only affected the Pitcairn Islands - population 47 people - and left the site stale for a week until JHU noticed and resolved the issue.

If we had a working alert system, we could use the population affected to adjust error handling.  We could define something akin to DEBUG/INFO/WARNING/ERROR thresholds, where data issues affecting populations above a certain size go to WARNING (actively alerting us) or even ERROR (where it would fail the prevalence run like it does today as well as alerting us).

Failing prevalence would represent a choice to leave the site with stale data rather than significantly limited data.